src/internal/proc.js

```diff
@@ -481,7 +485,17 @@ export default function proc(
 
     try {
       suspend()
-      const task = proc(taskIterator, subscribe, dispatch, getState, options, effectId, fn.name, (detached ? null : noop))
+      const task = proc(
+        taskIterator,
+        subscribe,
+        dispatch,
+        getState,
+        (detached ? null : taskContext), // TODO: topContext should be passed in case of spawn?
```

Comment:
Agree with @yelouafi, if we disable propagating context on `spawn`, it would make `context` a much less convenient mechanism for dependency injection.

That said, I think that `runSaga` should behave just like you said — starting with an empty context by default. Another reason is that otherwise we'd have to somehow link `runSaga` with `sagaMiddleware` instance.